src/Grpc.AspNetCore.HealthChecks/GrpcHealthChecksPublisher.cs

```diff
@@ -19,25 +19,44 @@
 using Grpc.Health.V1;
 using Grpc.HealthCheck;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
 
 namespace Grpc.AspNetCore.HealthChecks
 {
-    internal class GrpcHealthChecksPublisher : IHealthCheckPublisher
+    internal sealed class GrpcHealthChecksPublisher : IHealthCheckPublisher
     {
         private readonly HealthServiceImpl _healthService;
+        private readonly GrpcHealthChecksOptions _options;
 
-        public GrpcHealthChecksPublisher(HealthServiceImpl healthService)
+        public GrpcHealthChecksPublisher(HealthServiceImpl healthService, IOptions<GrpcHealthChecksOptions> options)
         {
             _healthService = healthService;
+            _options = options.Value;
         }
 
         public Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
         {
-            foreach (var entry in report.Entries)
+            foreach (var registration in _options.Services)
             {
-                var status = entry.Value.Status;
+                var filteredResults = report.Entries
+                    .Select(entry => new HealthResult(entry.Key, entry.Value.Tags, entry.Value.Status, entry.Value.Description, entry.Value.Duration, entry.Value.Exception, entry.Value.Data))
+                    .Where(registration.Predicate);
 
-                _healthService.SetStatus(entry.Key, ResolveStatus(status));
+                var resolvedStatus = HealthCheckResponse.Types.ServingStatus.Unknown;
```

Comment:
Nit: We can nuke the `ResolveStatus` method now since it doesn't appear to be used anywhere.